### PR TITLE
Replace image assets

### PR DESCRIPTION
**Pull request by arvida**
Makes it possible to change the file a asset points to. Only enabled for ImageAssets.

This makes it possible for other objects than AssetAssociations to be added as usages to Assets. This is needed so a asset can bust the cache for its usages it is updated.

### CODE REVIEW

**Comment by joeljunstrom** [comment]
This looks really nice! I'll make a proper read through as soon as I'm not on my phone :)

File: app/models/asset_usages.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  include Enumerable
+
+  attr_reader :asset
+
+  def initialize asset
+    @asset = asset
+  end
+
+  def each &block
+    using_objects.each {|using_object| yield using_object }
```
Comment (joeljunstrom) [inline app/models/asset_usages.rb]: Minor nitpick: Should be collapsable to  `using_objects.each &block` right?

**Comment by joeljunstrom** [comment]
I like the extraction into `AssetUsages` and I think most of the api makes sense. What strikes me as a bit strange / unexpected is the tight coupling between the asset and the asset usages instance.

`Asset` instantiates an `AssetUsages` and calls `#add obj`. The asses usage then formats the obj and mutates the assets state and tells the asset to persist itself.

To me it feels a bit awkward that AssetUsage not only can mutate state on asset but also persist it. Depending on what messages that same asset sends to it.

Looking at it maybe it would make more sense if AssetUsages knew how to be persisted to mongo ( by implementing `to_mongo` and `from_mongo` or use a transformator class) and that Asset implements AssetUsages as a field type?

File: app/models/asset_usages.rb (hunk 1 above)
Comment (joeljunstrom) [inline app/models/asset_usages.rb]: s/Should/Could/

**Comment by arvida** [comment]
Valid points, lets take a stab at a refactor for this later on.

**Comment by joeljunstrom** [comment]
:+1:

**Pull request merged**